taglocker_test.go

```diff
@@ -0,0 +1,162 @@
+package plc
+
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+	"testing"
+)
+
+const alphaCharset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+const numCharset = "0123456789"
+const alphanumCharset = alphaCharset + "_" + numCharset
+
+// NoOpReadWriter is a dummy ReadWriter interface that unconditionally succeeds.
+type NoOpReadWriter struct {
+	state uint32
+}
+
+func (norw NoOpReadWriter) ReadTag(name string, value interface{}) error {
+	garbage := norw.state
+	garbage++
+	return nil
+}
+func (norw NoOpReadWriter) WriteTag(name string, value interface{}) error {
+	norw.state++
+	return nil
+}
+
+// Generates a random fully-qualified tag, and returns
+// paths to that tag but also all intermediary prefixes.
+// We will test that we can lock intermediary nodes by
+// randomly choosing from these returned values.
+func genPaths() []string {
```

Comment:
I haven't reviewed this yet but I don't know if GitHub lets me save an in-progress review.
I also didn't look carefully at the structure of the benchmark function.